app/src/main/java/com/infomaniak/mail/ui/main/search/SearchFragment.kt

```diff
@@ -147,7 +147,24 @@ class SearchFragment : TwoPaneFragment() {
     }
 
     private fun setupAdapter() {
-        threadListAdapter(folderRole = null, isFolderNameVisible = true)
+        threadListAdapter(
+            folderRole = null,
+            isFolderNameVisible = true,
+            onThreadClicked = { thread ->
+                with(searchViewModel) {
+                    if (!isLengthTooShort(currentSearchQuery)) history.value = currentSearchQuery
+                    binding.searchBar.searchTextInput.apply {
+                        hideKeyboard()
+                        clearFocus()
+                    }
+                    navigateToThread(thread)
+                }
+            },
+        )
+
+        binding.mailRecyclerView.adapter = threadListAdapter.apply {
+            stateRestorationPolicy = StateRestorationPolicy.PREVENT_WHEN_EMPTY
+        }
```

Comment:
Maybe keep the `mailRecyclerView.adapter = threadListAdapter` with the other setMessageUi function